react-native-share/react-native-share
# chore(example): update example to rn66

# Overview

fixes android build breaks (jcenter is super flaky now, react-native 66 doesn't need it anymore) and fixes ios build breaks (react-native < 0.64.2 does not even build on Xcode > 12.4)

# Test Plan

1. Run example app locally (`yarn && yarn start` in one terminal, `yarn start:android` and/or `yarn start:ios` in another terminal)
1. see the same build error seen in #1140 
1. use upgrade-helper to update from react-native 0.63.x to 0.66.x
1. re-run the example app locally, see it pass
